### PR TITLE
Component/badge

### PR DESCRIPTION
**Pull request by mach33n**
Tracking issue: #11 

## Summary
This PR introduces the fl-badge to the library. It currently possesses 3 attributes and some displayed test cases in the index.html.

## Tests

Noteworthy cases:
Please see fl-badge.spec.tsx, it has examples of the badge in its base state without any props. In addition, feel free to check out the earlier mentioned index.html file for examples of using the badge component.

## Documentation
# fl-badge

## Properties

| Property     | Attribute    | Description | Type                                                                                                         | Default          |
| ------------ | ------------ | ----------- | ------------------------------------------------------------------------------------------------------------ | ---------------- |
| `pill`       | `pill`       | `This attribute rounds the edges of desired badge and gives it the look of a pill.`            | `boolean`                                                                                                    | `false`          |
| `position`   | `position`   | `This attribute pushes the pill slightly over the desired corner on its parent element.`            | `BOTTOMLEFT \| BOTTOMRIGHT \| NONE \| TOPLEFT \| TOPRIGHT` | `NONE` |
| `type`       | `type`       | `These badge types adjust text defaults and background colors for the desired badge.`            | `DANGER \| INFO \| PRIMARY \| SUCCESS \| WARNING`                         | `PRIMARY` |

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*

### CODE REVIEW

File: packages/core/src/components/fl-badge/fl-badge.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+}
+render() {
+  return (
+    <Host 
+      class={{
+        "badge": true,
+        [`badge--${this.type}`]: true,
+        [`badge--${this.position}`]: true,
+        "badge-pill": this.pill,
+      }}> {!this.element.innerHTML ? this.type : null}
```
Comment (1ntEgr8) [inline packages/core/src/components/fl-badge/fl-badge.tsx]: Do we want to render the type of the badge always? I think it'll be nice to customize this.

I think you can safely omit the `{!this.element.innerHTML ? this.type : null}`. The slot should be able to handle the rest

**Review by 1ntEgr8 — CHANGES_REQUESTED** [review]
Overall, it looks good.

Can you merge in the changes from master? You can move your example code from `index.html` to the corresponding markdown file for badge. 11ty will render it correctly.

File: packages/core/src/components/fl-badge/fl-badge.tsx (hunk 1 above)
Comment (mach33n) [reply packages/core/src/components/fl-badge/fl-badge.tsx]: Just made an updated branch with merges from main.

File: packages/core/src/components/fl-badge/fl-badge.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+})
+export class FlBadge {
+
+@Prop() type: btypes = btypes.PRIMARY;
+
+@Prop() pill = false;
+
+@Prop() position: positions = positions.NONE;
+
+/* stylelint-disable */
```
Comment (1ntEgr8) [inline packages/core/src/components/fl-badge/fl-badge.tsx]: Why is there a `stylelint-disable` comment?

File: packages/core/src/components/fl-badge/fl-badge.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+@Element() private element: HTMLFlBadgeElement;
+/* stylelint-enable */
+
+componentWillLoad() {
+  this.element.parentElement.style.position = "relative";
+  // Just ensures some default type is set for badge
```
Comment (1ntEgr8) [inline packages/core/src/components/fl-badge/fl-badge.tsx]: you can set default types in the declaration. you don't need this

File: packages/core/src/components/fl-badge/fl-badge.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+@Prop() pill = false;
+
+@Prop() position: positions = positions.NONE;
+
+/* stylelint-disable */
+@Element() private element: HTMLFlBadgeElement;
+/* stylelint-enable */
+
+componentWillLoad() {
+  this.element.parentElement.style.position = "relative";
```
Comment (1ntEgr8) [inline packages/core/src/components/fl-badge/fl-badge.tsx]: you can set the position in CSS. add it to the :host selector or a container

File: packages/core/src/components/fl-badge/fl-badge.scss [hunk 5]
```diff
@@ -0,0 +1,64 @@
+:host {
+  display: block;
+
```
Comment (1ntEgr8) [inline packages/core/src/components/fl-badge/fl-badge.scss:3]: you can add the `position: relative` here

**Review by 1ntEgr8 — CHANGES_REQUESTED** [review]
- delete the `package-lock.json` file. we are using `yarn`
- remove the `swp` files (see comment in #32)
- remove the `index.html` file. Add your examples to the readme. Take a look at how this is done in the `main` branch

There is more to be desired from the styling. I'm a little concerned that the badge placement isn't generalized. We'll need to test it

File: packages/core/src/components/fl-badge/fl-badge.tsx (hunk 2 above)
Comment (mach33n) [reply packages/core/src/components/fl-badge/fl-badge.tsx]: I kept getting a lint error that said I could only declare HTMLElement as an appropriate type but I needed to use the autogenerated HTMLBadgeElement in order to reference the self element properly.

Comment (mach33n) [reply packages/core/src/components/fl-badge/fl-badge.tsx]: I just removed it thanks to your later comments.

File: packages/core/src/components/fl-badge/fl-badge.scss [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  --fl-badge-border-radius: 16px;
+  --fl-badge-position: absolute;
```
Comment (1ntEgr8) [inline packages/core/src/components/fl-badge/fl-badge.scss:5]: Not sure if we need to make this configurable. But not a big deal

**Review by 1ntEgr8 — APPROVED** [review]

**Pull request merged**